clang/lib/Driver/Driver.cpp

```diff
@@ -1556,6 +1556,32 @@ static void appendOneArg(InputArgList &Args, const Arg *Opt) {
   }
 }
 
+// Utility function to parse all devices passed via -fsycl-targets.
+// Return 'true' for JIT, AOT Intel CPU/GPUs and NVidia/AMD targets.
+// Otherwise return 'false'.
+bool Driver::GetUseNewOffloadDriverForSYCLOffload(Compilation &C,
+                                                  const ArgList &Args) const {
+  // Check only if enabled with -fsycl
+  if (!Args.hasFlag(options::OPT_fsycl, options::OPT_fno_sycl, false))
+    return false;
+
+  if (Args.hasFlag(options::OPT_no_offload_new_driver,
+                   options::OPT_offload_new_driver, false))
+    return false;
+
+  if (Args.hasArg(options::OPT_fintelfpga))
```

Comment:
Support for FPGA related options have been removed in this [PR.](https://github.com/intel/llvm/pull/16864)
Don't think this check is necessary.